nifi-nar-bundles/nifi-standard-services/nifi-hbase_1_1_2-client-service-bundle/nifi-hbase_1_1_2-client-service/src/main/java/org/apache/nifi/hbase/HBase_1_1_2_LookupService.java

```diff
@@ -0,0 +1,168 @@
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.Validator;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.controller.ControllerServiceInitializationContext;
+import org.apache.nifi.lookup.LookupFailureException;
+import org.apache.nifi.lookup.LookupService;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.reporting.InitializationException;
+import org.apache.nifi.serialization.SimpleRecordSchema;
+import org.apache.nifi.serialization.record.MapRecord;
+import org.apache.nifi.serialization.record.RecordField;
+import org.apache.nifi.serialization.record.RecordFieldType;
+import org.apache.nifi.serialization.record.RecordSchema;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.Optional;
+import java.util.Set;
+
+public class HBase_1_1_2_LookupService extends HBase_1_1_2_ClientService implements LookupService {
+    private static final Set<String> REQUIRED_KEYS;
+
+    public static final PropertyDescriptor TABLE_NAME = new PropertyDescriptor.Builder()
+            .name("hb-lu-table-name")
+            .displayName("Table Name")
+            .description("The name of the table where look ups will be run.")
+            .required(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor RETURN_CFS = new PropertyDescriptor.Builder()
+            .name("hb-lu-return-cfs")
+            .displayName("Column Families")
+            .description("The column families that will be returned.")
+            .required(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor RETURN_QFS = new PropertyDescriptor.Builder()
+            .name("hb-lu-return-qfs")
+            .displayName("Column Qualifiers")
+            .description("The column qualifies that will be returned.")
+            .required(false)
+            .addValidator(Validator.VALID)
+            .build();
+
+    static {
+        REQUIRED_KEYS = new HashSet<String>();
+        REQUIRED_KEYS.add("rowKey");
+    }
+
+    private String tableName;
+    private List<byte[]> families;
+    private List<byte[]> qualifiers;
+
+    private List<PropertyDescriptor> lookupProperties;
+
+    @Override
+    protected void init(ControllerServiceInitializationContext config) throws InitializationException {
+        super.init(config);
+        this.lookupProperties = new ArrayList<>();
+        this.lookupProperties.addAll(properties);
+        this.lookupProperties.add(TABLE_NAME);
+        this.lookupProperties.add(RETURN_CFS);
+        this.lookupProperties.add(RETURN_QFS);
+    }
+
+    @Override
+    public Optional lookup(Map coordinates) throws LookupFailureException {
```

Comment:
This should be declared as returning `Optional<Record>` and take `Map<String, String>` as the argument.